### PR TITLE
Attachment forwarding

### PR DESCRIPTION
**Pull request by GAMELASTER**
You can forward various attachments for multiple users.
Actually I tested videos and images. GIFs should work too.
Can anyone test forwading files and audio files?

### CODE REVIEW

**Comment by ravkr** [comment]
wow, i'm impressed! :smile:

File: src/forwardAttachment.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    var timestamp = Math.floor(Date.now() / 1000);
+
+    for (var i = 0; i < userOrUsers.length; i++) {
+      //That's good, the key of the array is really timestmap in seconds + index
+      //Probably time when the attachment will be sent?
+      form['recipient_map[' + (timestamp + i) + ']'] = userOrUsers[i];
+    }
+
+    defaultFuncs
+      .post("https://www.messenger.com/mercury/attachments/forward/?dpr=1", ctx.jar, form)
```
Comment (bsansouci) [inline src/forwardAttachment.js]: I would imagine you can remove that `?dpr=1` param. Could you double check if it works without?

File: DOCS.md [hunk 2]
```diff
@@ -412,6 +413,18 @@ login({appState: JSON.parse(fs.readFileSync('appstate.json', 'utf8'))}, (err, ap
 
 ---------------------------------------
 
+<a name="forwardAttachment"></a>
+### api.forwardAttachment(attachmentID, userOrUsers[, callback])
+
+Takes a userID or an array of userIDs and forward the corresponding attachment.
+
+__Arguments__
+* `attachmentID`: A attachment ID (usualy `attribute.ID` property)
```
Comment (bsansouci) [inline DOCS.md]: Could you rephrase this to be:

The `ID` field in the attachment object. Not all attachment have `ID`s: recorded audio and arbitrary files don't for example.

**Comment by bsansouci** [comment]
It seems that I cannot forward audio files or files from the UI. So it's ok to not cover it (we don't get `ID` from them).

I added some comments. I can merge once you've fixed them.

File: src/forwardAttachment.js (hunk 1 above)
Comment (GAMELASTER) [inline src/forwardAttachment.js]: Removed, It's works good without it

File: DOCS.md (hunk 2 above)
Comment (GAMELASTER) [inline DOCS.md]: Thank you a lot for this, I still have problems with describing arguments (Or anything other). My English vocabulary and grammar is still bad.

**Comment by GAMELASTER** [comment]
Thank you a lot for comments! I solved all of them!

File: DOCS.md (hunk 2 above)
Comment (Schmavery) [inline DOCS.md]: Hey, you're doing great! It's a tricky thing to get perfect.
(Don't worry, @bsansouci's first language is French so he understands how you feel)

Comment (ravkr) [inline DOCS.md]: To add my few cents: my first language is Polish :smile:

File: DOCS.md [hunk 3]
```diff
@@ -412,6 +413,18 @@ login({appState: JSON.parse(fs.readFileSync('appstate.json', 'utf8'))}, (err, ap
 
 ---------------------------------------
 
+<a name="forwardAttachment"></a>
+### api.forwardAttachment(attachmentID, userOrUsers[, callback])
+
+Takes a userID or an array of userIDs and forward the corresponding attachment.
```
Comment (ravkr) [inline DOCS.md]: I would rephrase it this way:
"Forwards corresponding attachment to given userID or to every user from an array of userIDs"

Comment (GAMELASTER) [inline DOCS.md]: Done

**Comment by bsansouci** [comment]
Awesommmmeee! Merging!
